.github/workflows/xcmetrics.yml

```diff
@@ -0,0 +1,76 @@
+name: Performance Benchmarks
+
+on:
+  schedule:
+    # Runs "At 03:00 every night"
+    - cron: '0 3 * * *'
+
+  pull_request_review:
+    types: [submitted, edited]
+
+  workflow_dispatch:
+
+env:
+  HOMEBREW_NO_INSTALL_CLEANUP: 1 # Disable cleanup for homebrew, we don't need it on CI
+
+jobs:
+  xcmetrics:
+    name: XCMetrics
+    runs-on: macos-13
+    if: ${{ github.event_name != 'pull_request_review' || github.event.review.state == 'approved' }}
+    steps:
+      - name: Install Bot SSH Key
+        uses: webfactory/ssh-agent@v0.7.0
+        with:
+          ssh-private-key: ${{ secrets.BOT_SSH_PRIVATE_KEY }}
+
+      - name: Check for PR approvals
+        if: ${{ github.event_name == 'pull_request_review' && github.event.review.state == 'approved' }}
+        env:
+          GITHUB_TOKEN: '${{ secrets.CI_BOT_GITHUB_TOKEN }}'
+        run: |
+          current_branch=${{ github.event.pull_request.head.ref }}
+          echo "Current branch: $current_branch"
+          approvals_count=$(gh api "https://api.github.com/repos/GetStream/stream-chat-swift/pulls/${{ github.event.pull_request.number }}/reviews" | jq '[.[] | select(.state == "APPROVED")] | length')
```

Comment:
Thinking if it would be nicer if we pull out these URLs and have them as ENV variables or some sort of a constants. But on the other hand, does not really matter much since we do not expect these to change anyway.

Reply:
Yeah, don't see a reason to extract them into a separate env variable as they will only be used in this place. But we could extract the whole step into an independent action. It will be just a bit more readable 🤔

Will think about it 👍

Reply:
@laevandus, I extracted it into a separate action. Wdyt?

Reply:
The only thing is that to test this, we need to approve the PR 😄
Will test it then when everything else is reviewed/approved.

Reply:
@laevandus, I need another round of testing, could you approve one more time please?

Reply:
Done ✅